text/0308-cli-advisories.md

```diff
@@ -0,0 +1,280 @@
+
+### Why do we need to give users the option to acknowledge advisories?
+
+Given the expected lifetime of advisories, they will eventually become just
+noise and users will try to silence it. If the only option we give users is to
+suppress them (per execution or indefinitely), they will use that and miss new
+advisories.
+
+### What future improvements can we make?
+
+**Resource filter**: Ideally, advisories should be as targeted as possible. The
+version filter addresses part of that requirement, but we can make it better by
+also adding a resource filter: if an issue only affects `FargateTaskDefinition`
+resources, for example, but this resource is not present in any of the stacks,
+the customer doesn’t need to see that particular warning.
+
+**Pipeline awareness**: Initially, we won’t treat the pipeline as a special
+case. By default, the advisories will be fetched and displayed and we expect
+users to never look at them. This behavior can always be overridden by
+suppressing the advisories. In future releases, the CLI will use the information
+about whether it’s running on a pipeline and block the deployment in case there
+is any advisory.
+
+## Appendix
+
+### Detailed design
+
+#### Publishing advisories
+
+We will create a new repository, `cdklabs/aws-cdk-advisories`, dedicated to host
+the advisories file. As usual, in order to be merged, any change to this file
+will have to be published as a PR and approved. Using a CODEOWNERS file, we will
+restrict the permission to approve PRs only to SDMs. The file will contain a
+list of advisories, each having the following fields:
+
+|     Field    |                           Description                          | Format                          | Mandatory? |
+|:------------:|:--------------------------------------------------------------:|---------------------------------|:----------:|
+| `title`      | The title of the incident                                      | Free form text                  | Yes        |
+| `issueUrl`   | A link to the GitHub issue where the incident is being tracked | URL                             | Yes        |
+| `overview`   | A paragraph with more information about the incident           | Free form text                  | Yes        |
+| `component`  | The CLI or the Framework                                       | Either `"cli"` or `"framework"` | Yes        |
+| `version`    | Version range using the semver format                          | Semantic Versioning             | No         |
+
+
+We will also implement three GitHub actions on this repository:
+
+1. File validation on PR. It will block merging if the structure of the file is
+   not compliant with the specification above.
+2. Issue sync-up. When the PR is merged, this action will copy the content of
+   the file over to the GitHub issue it's linked to.
+3. Issue protection. Every change to issues that are linked to some advisory
+   will be checked by this action, to avoid corruption.
```

Comment:
What's the difference between 2 and 3? I'd assume it's a workflow that just iterates over all the advisories and syncs up their issues. No?